src/specBuilder/trendline/trendlineSignalUtils.ts

```diff
@@ -9,25 +9,52 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-import { hasTooltip } from '@specBuilder/marks/markUtils';
+import { RSC_ANIMATION } from '@constants';
+import { TrendlineSpecProps } from '@rsc';
+import { hasPopover, hasTooltip } from '@specBuilder/marks/markUtils';
 import {
 	addHighlightedItemSignalEvents,
 	addHighlightedSeriesSignalEvents,
+	getRscAnimationSignals,
+	getRscTrendlineColorAnimationDirection,
+	hasSignalByName,
 } from '@specBuilder/signal/signalSpecBuilder';
 import { Signal } from 'vega';
 
 import { TrendlineParentProps, getTrendlines } from './trendlineUtils';
 
 export const setTrendlineSignals = (signals: Signal[], markProps: TrendlineParentProps): void => {
-	const { name: markName } = markProps;
+	const { name: markName, animations } = markProps;
 	const trendlines = getTrendlines(markProps);
-
+	// if animations are enabled, add necessary opacity animation signals
+	//TODO: Add tests
+	if (animations) {
+		checkRSCTrendlineAnimationSignals(markName, signals, trendlines);
```

Comment:
This function name could make it more clear that we're mutating the data. Maybe `add` or `push` instead of `check`.